src/org/starexec/data/security/SpaceSecurity.java

```diff
@@ -650,7 +650,7 @@ public static ValidatorStatusCode canCopyOrLinkSolverBetweenSpaces(
 		} else {
 			for (Integer sid : solverIdsBeingCopied) {
 				Solver solver = Solvers.get(sid);
-				if (!SolverSecurity.userOwnsSolverOrIsAdmin(solver, userId)) {
+				if (!SolverSecurity.userOwnsSolverOrIsAdmin(solver, userId) && !Permissions.canUserSeeSolver(solver.getId(), userId)) {

```

Comment:
Instead of `Permissions.canUserSeeSolver` should this be `SolverSecurity.canUserDownloadSolver`?